src/helics/core/ActionMessage.cpp

```diff
@@ -189,7 +189,7 @@ std::string ActionMessage::packetize () const
     // don't forget to flush the stream to finish writing into the buffer
     s.flush ();
     // now generate a length header
-    auto sz = static_cast<int32_t> (data.size ());
+    auto sz = static_cast<uint32_t> (data.size ());
```

Comment:
Why are we using unsigned here? From Google's cpp style guide: 

> You should not use the unsigned integer types such as uint32_t, unless there is a valid reason such as representing a bit pattern rather than a number, or you need defined overflow modulo 2^N. In particular, do not use unsigned types to say a number will never be negative. Instead, use assertions for this. [1]

[1] - https://google.github.io/styleguide/cppguide.html#Integer_Types

Reply:
the next lines are bit shifts and & operations.    clang-tidy flag it as a code issue since you are doing bit operations and packing a raw data structure.

Reply:
and that should be done on unsigned values

Reply:
Okay great!